src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -261,28 +262,15 @@ const ProxyDeviceForm: FC<Props> = ({ formik, project }) => {
         className: "no-border-top inherited-with-form",
         configuration: <Label>Listen</Label>,
         inherited: (
-          <Input
-            name={`devices.${index}.listen`}
-            id={`devices.${index}.listen`}
-            key={`devices.${index}.listen`}
-            onBlur={formik.handleBlur}
-            onChange={(e) => {
+          <ProxyDeviceInput
+            onChange={(type, address, port) => {
               ensureEditMode(formik);
               void formik.setFieldValue(
                 `devices.${index}.listen`,
-                e.target.value,
+                `${type}:${address}:${port}`,
```

Comment:
If nat mode is enabled, we should also update the connect type from here. I think we could do as below 
```
const connectParts = formDevice.connect.split(":");
const connectAddress = connectParts.length > 1 ? connectParts[1] : "";
const connectPort = connectParts.length > 2 ? connectParts[2] : "";
void formik.setFieldValue(
                `devices.${index}.connect`,
                e.target.value,
                `${type}:connectAddress:connectPort`,
```